crates/op-rbuilder/src/flashtestations/builder_tx.rs

```diff
@@ -0,0 +1,581 @@
+                return true;
+            }
+        }
+        false
+    }
+
+    fn fund_tee_service_tx<ExtraCtx: Debug + Default>(
+        &self,
+        ctx: &OpPayloadBuilderCtx<ExtraCtx>,
+        evm: &mut OpEvm<
+            &mut State<StateProviderDatabase<impl StateProvider>>,
+            NoOpInspector,
+            PrecompilesMap,
+        >,
+    ) -> Result<Option<BuilderTransactionCtx>, BuilderTransactionError> {
+        let balance = get_balance(evm.db_mut(), self.tee_service_signer.address)?;
+        if balance.is_zero() {
+            let funding_nonce = get_nonce(evm.db_mut(), self.funding_key.address)?;
+            let funding_tx = self.signed_funding_tx(
+                self.tee_service_signer.address,
+                self.funding_key,
+                self.funding_amount,
+                ctx.base_fee(),
+                ctx.chain_id(),
+                funding_nonce,
+            )?;
+            let da_size =
+                op_alloy_flz::tx_estimated_size_fjord_bytes(funding_tx.encoded_2718().as_slice());
+            let ResultAndState { state, .. } = match evm.transact(&funding_tx) {
+                Ok(res) => res,
+                Err(err) => {
+                    if err.is_invalid_tx_err() {
+                        warn!(target: "flashtestations", %err, "funding tx failed");
+                        return Ok(None);
+                    } else {
+                        return Err(BuilderTransactionError::EvmExecutionError(Box::new(err)));
+                    }
+                }
+            };
+            info!(target: "flashtestations", block_number = ctx.block_number(), tx_hash = ?funding_tx.tx_hash(), "adding funding tx to builder txs");
+            evm.db_mut().commit(state);
+            Ok(Some(BuilderTransactionCtx {
+                gas_used: 21000,
+                da_size,
+                signed_tx: funding_tx,
+                is_top_of_block: true,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn register_tee_service_tx<ExtraCtx: Debug + Default>(
+        &self,
+        ctx: &OpPayloadBuilderCtx<ExtraCtx>,
+        evm: &mut OpEvm<
+            &mut State<StateProviderDatabase<impl StateProvider>>,
+            NoOpInspector,
+            PrecompilesMap,
+        >,
+    ) -> Result<(Option<BuilderTransactionCtx>, bool), BuilderTransactionError> {
+        let TxSimulateResult {
+            gas_used,
+            success,
+            state_changes,
+            revert_reason,
+            logs,
+        } = self.simulate_register_tee_service_tx(ctx, evm)?;
+        if success {
+            if !self.check_tee_address_registered_log(&logs, self.tee_service_signer.address) {
+                warn!(target: "flashtestations", "transaction did not emit TEEServiceRegistered log, FlashtestationRegistry contract address may be incorrect");
+                Ok((None, false))
```

Comment:
not propagating the error here seems like it could get messy, why not propagate the errors and handle them in the caller?